### DISCUSSION TITLE
Accessing "app-thinning.plist" after a build using `gym`?

### DISCUSSION THREAD

**Discussion by dhoskins** [fastlane/fastlane#17374] [Q&A]
After exporting an archive manually in Xcode, a folder gets created which contains files like `App Thinning Size Report.txt` and `app-thinning.plist`. 

Is it possible to access and surface these artefacts after a `gym` build in order to publish the size somewhere? 

This is particularly in relation to keeping track of the size of an App Clips project.

**Comment by rogerluan** [comment]
That's a great question and I'd love to see that as well.

Have you tried SwiftInfo @dhoskins ? https://github.com/rockbruno/SwiftInfo

Let me know if that suits your needs.

If not, if you'd like to have that report from xcodebuild, I think you'd need to export for ad hoc (just like you're doing in Xcode UI probably). I've never tried this myself though, so not sure if it's as straight forward as it seems, or where that report would be located at 😅 definitely needs some digging.

**Comment by dhoskins** [comment]
Yes it's the app-thinning report I want. I think you're right that an ad-hoc build would be required, in fact I have just noticed here https://developer.apple.com/documentation/xcode/reducing_your_app_s_size that it says:

> Export your archived app as an Ad Hoc, Development, or Enterprise build.

Which is unfortunate as I am doing an App Clip and the only way to test that is on Test Flight, so it's App Store exports I've been doing! I'll try enabling this feature on an Ad Hoc build tomorrow and see how I get on.

**Comment by rogerluan** [comment]
Alternatively, you could build for App Store, and ad-hoc - the later just to get the report (if it's possible) 😄

**Reply by dhoskins** [reply]
Building for ad hoc does generate a report, but unfortunately it does not include details for an App Clip product. I can't find a way to generate that at the moment other than through Xcode.

**Reply by rogerluan** [reply]
Ahh 😞 I understand now. This probably needs to some digging as it's a new Xcode 12 capability (might not be documented yet).
If you find anything related to this, let us know, so we can track it and maybe include in _fastlane_ 🙏

**Comment by asoneji** [comment]
Any suggestion to `App Thinning Size Report.txt` after a build using `gym`?

**Reply by asoneji** [reply]
after adding https://docs.fastlane.tools/actions/build_ios_app/#export-options to my gym options

```
    export_options: ({
        thinning: "<thin-for-all-variants>"
    })
```

It generated `App Thinning Size Report.txt` and `app-thinning.plist`
